### PR TITLE
add support for cuda9 on windows test=develop

### PR DESCRIPTION
**Pull request by wopeizl**
add a patch for eigen since eigen has a bug for cuda9 on windows
add cuda binary adaption

### CODE REVIEW

**Review by qingqing01 — COMMENTED** [review]
LGTM.

**Review by qingqing01 — COMMENTED** [review]
LGTM.

File: cmake/external/eigen.cmake [hunk 1]
```diff
@@ -12,6 +12,13 @@ if(NOT WITH_FAST_MATH)
   add_definitions(-DEIGEN_FAST_MATH=0)
 endif()
 
+if(WIN32)
+    set(PATCH_COMMAND ${CMAKE_COMMAND} -E copy_if_different
+            ${PADDLE_SOURCE_DIR}/patches/eigen/Half.h.txt
+            ${EIGEN_INCLUDE_DIR}/Eigen/src/Core/arch/CUDA/Half.h)
```
Comment (chengduoZH) [inline cmake/external/eigen.cmake]: **I don't think we should do that for the users who need to compile Paddle in Windows, we should let users do that themselves to modify this file.**

Comment (wopeizl) [reply cmake/external/eigen.cmake]: remove the patch

**Review by qingqing01 — APPROVED** [review]
LGTM.

**Pull request merged**